pkg/pool-weighted/test/foundry/LiquidityApproximationWeighted.t.sol

```diff
@@ -126,159 +164,286 @@ contract LiquidityApproximationWeightedTest is LiquidityApproximationTest {
         uint256 swapFeePercentage,
         uint256 weightDai
-    ) internal {
+    ) public {
         // Weights can introduce some differences in the swap fees calculated by the pool during unbalanced add/remove
         // liquidity, so the error tolerance needs to be a bit higher than the default tolerance.
         excessRoundingDelta = 0.5e16; // 0.5%
 
-        swapFeePercentage = _setPoolWeightsAndSwapFee(swapFeePercentage, weightDai);
+        _setPoolBalancesWithDifferentWeights(weightDai);
 
         uint256 amountOut = removeExactOutAllUsdcAmountOut(exactBptAmountOut, swapFeePercentage);
         swapFeePercentage > 0
             ? assertLiquidityOperation(amountOut, swapFeePercentage, false)
             : assertLiquidityOperationNoSwapFee();
     }
 
+    // #endregion addProportionalRemoveExactOut
+
+    // #region removeExactOut
+
     function testRemoveLiquiditySingleTokenExactOutWeights__Fuzz(
         uint256 exactAmountOut,
         uint256 swapFeePercentage,
         uint256 weightDai
     ) public {
-        _removeLiquiditySingleTokenExactOutWeights(exactAmountOut, swapFeePercentage, weightDai);
+        // This test adds 10x the initial liquidity to work, so we amplify the usual min amount.
+        // On the other hand, we would need to add even more in the first step to work with large `exactAmountOut`,
+        // so we also cap the maximum.
+        exactAmountOut = bound(exactAmountOut, minAmount * 10, maxAmount / 10);
+        swapFeePercentage = bound(swapFeePercentage, minSwapFeePercentage, maxSwapFeePercentage);
+        weightDai = bound(weightDai, 1e16, 99e16);
+
+        removeLiquiditySingleTokenExactOutWeights(exactAmountOut, swapFeePercentage, weightDai);
     }
 
     function testRemoveLiquiditySingleTokenExactOutWeightsNoSwapFee__Fuzz(
         uint256 exactAmountOut,
         uint256 weightDai
     ) public {
-        _removeLiquiditySingleTokenExactOutWeights(exactAmountOut, 0, weightDai);
+        // This test adds 10x the initial liquidity to work, so we amplify the usual min amount.
+        // On the other hand, we would need to add even more in the first step to work with large `exactAmountOut`,
+        // so we also cap the maximum.
+        exactAmountOut = bound(exactAmountOut, minAmount * 10, maxAmount / 10);
+        weightDai = bound(weightDai, 1e16, 99e16);
+
+        removeLiquiditySingleTokenExactOutWeights(exactAmountOut, 0, weightDai);
+    }
+
+    function testRemoveLiquiditySingleTokenExactOutWeightsSmallAmounts__Fuzz(
+        uint256 exactAmountOut,
+        uint256 weightDai
+    ) public {
+        exactAmountOut = bound(exactAmountOut, 1, 1e6);
+        weightDai = bound(weightDai, 1e16, 99e16);
+
+        try this.removeLiquiditySingleTokenExactOutWeights(exactAmountOut, 0, weightDai) {
+            // OK, test passed.
+        } catch (bytes memory result) {
+            assertEq(bytes4(result), bytes4(stdError.arithmeticError), "Unexpected error");
```

Comment:
Would also add a comment here, like "Can legitimately fail due to arithmetic underflow; live system will be protected by minimum amounts." Could even point to the line where it would be expected to fail, if we're close enough that we don't expect maintenance issues with pointing to specific lines.

Reply:
You have a similar comment below at L397: 

  // For very small invariant ratios, `BasePoolMath` reverts when calculated amount out < 0 because of rounding.
  // Perform an external call so that `expectRevert` catches the error.